pkg/plugin/dropreason/ebpfsetup_linux.go

```diff
@@ -0,0 +1,204 @@
+package dropreason
+
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/blang/semver/v4"
+	"github.com/cilium/cilium/pkg/version"
+	"github.com/cilium/cilium/pkg/versioncheck"
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/link"
+	plugincommon "github.com/microsoft/retina/pkg/plugin/common"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+const (
+	MinAmdVersionNum = "5.5"
+	MinArmVersionNum = "6.0"
+)
+
+/*
+getEbpfPayload() returns the ebpf program and map objects to load, based on the kernel version, architecture, and distro.
+We use fexit programs for better performance, and fall back to kprobes.
+
+eBPF Program Support Matrix
+===========================
+
+Program Type Selection:
+- If:
+  - Arch == amd64 and kernel >= 5.5
+  - Arch == arm64 and kernel >= 6.0
+    → Use `fexit`
+  - Else:
+    → Use `kprobe`
+
+Scope Selection:
+- If:
+  - Distro == Mariner
+    → Scope = core (only core kernel funcs)
+  - Else:
+    → Scope = all (core + module funcs)
+
++-----------+------------------------+--------------+--------+
+| Distro    | Arch + Kernel          | Prog         | Scope  |
++-----------+------------------------+--------------+--------+
+| Mariner   | amd64, kernel >= 5.5   | fexit        | core   |
+| Mariner   | arm64, kernel >= 6.0   | fexit        | core   |
+| Non-Marin | amd64, kernel >= 5.5   | fexit        | all    |
+| Non-Marin | arm64, kernel >= 6.0   | fexit        | all    |
+| *         | (otherwise)            | kprobe       | per OS |
++-----------+------------------------+--------------+--------+
+
+core kernel funcs:
+- tcp_v4_connect
+- inet_csk_accept
+- nf_hook_slow
+
+module funcs:
+- nf_conntrack_confirm
+- nf_nat_inet_fn
+*/
+
+func (dr *dropReason) getEbpfPayload() (objs interface{}, maps *kprobeMaps, isFexit bool, err error) {
+	isMariner := plugincommon.IsAzureLinux()
+	dr.l.Info("Distro check:", zap.Bool("isMariner", isMariner))
+
+	kv, err := version.GetKernelVersion()
+	if err != nil {
+		dr.l.Warn("Failed to get kernel version", zap.Error(err))
```

Comment:
This is not needed if fallback `GetKernelVersionMajMin` succeeds.

Reply:
removed